### PR TITLE
Add support for nd-vector

### PR DESCRIPTION
**Pull request by leonardt**
For code generating ndarrays

Also allows nested index nodes (for indexing the ndarrays)

### CODE REVIEW

File: tests/basic.cpp [hunk 1]
```diff
@@ -122,6 +122,15 @@ TEST(BasicTests, TestVector) {
   EXPECT_EQ(slice.toString(), "[31:0] x");
 }
 
+TEST(BasicTests, TestNDVector) {
+  std::vector<std::unique_ptr<vAST::Expression>> dims;
+  dims.push_back(vAST::make_num("8"));
+  dims.push_back(vAST::make_num("32"));
+  vAST::NDVector slice(vAST::make_id("x"), vAST::make_num("31"),
+                       vAST::make_num("0"), std::move(dims));
+  EXPECT_EQ(slice.toString(), "[31:0] x [8][32]");
```
Comment (leonardt) [inline tests/basic.cpp]: Don't remember exactly what the desired multi-dimensional array syntax is, but this is easy enough to change or make different options for.

File: include/verilogAST.hpp [hunk 2]
```diff
@@ -383,11 +383,13 @@ class Concat : public Expression {
 
  public:
   std::vector<std::unique_ptr<Expression>> args;
+  bool unpacked;
```
Comment (rsetaluri) [inline include/verilogAST.hpp:386]: what does this flag control? seems like it inserts a `'`? not familiar with this in verilog.

File: include/verilogAST.hpp [hunk 3]
```diff
@@ -222,31 +222,19 @@ class Slice : public Expression {
...
  public:
   std::variant<std::unique_ptr<Identifier>, std::unique_ptr<Attribute>,
-               std::unique_ptr<Slice>>
+               std::unique_ptr<Slice>, std::unique_ptr<Index>>
       value;
   std::unique_ptr<Expression> index;
 
   Index(std::variant<std::unique_ptr<Identifier>, std::unique_ptr<Attribute>,
-                     std::unique_ptr<Slice>>
+                     std::unique_ptr<Slice>, std::unique_ptr<Index>>
```
Comment (rsetaluri) [inline include/verilogAST.hpp:237]: why is this necessary for this change?

File: tests/basic.cpp (hunk 1 above)
Comment (rsetaluri) [reply tests/basic.cpp]: Think this way is good for now (believe that was the FB feedback as well)

File: include/verilogAST.hpp (hunk 2 above)
Comment (leonardt) [reply include/verilogAST.hpp:386]: `{x, y}` concatenates the bits of x and y together, `'{x, y}` will make it a 2-d unpacked array of bits.

File: include/verilogAST.hpp (hunk 3 above)
Comment (leonardt) [reply include/verilogAST.hpp:237]: For multi-dimensional arrays, we need to nest index nodes (e.g. `x[0][1]`), so an Index node needs to be able to have an Index as it's value

File: include/verilogAST.hpp (hunk 2 above)
Comment (leonardt) [reply include/verilogAST.hpp:386]: Here's an example https://verificationacademy.com/forums/systemverilog/unpacked-array-concatenation-assignment

**Review by rsetaluri — APPROVED** [review]
